modules/renter/renter.go

```diff
@@ -455,6 +463,22 @@ func NewCustomRenter(g modules.Gateway, cs modules.ConsensusSet, tpool modules.T
 		return nil, err
 	}
 
+	// Set RenterSettings to persisted data
+	if r.persist.MaxdownloadSpeed == 0 {
+		r.persist.MaxdownloadSpeed = defaultMaxDownloadSpeed
+	}
+	if r.persist.MaxUploadSpeed == 0 {
+		r.persist.MaxUploadSpeed = defaultMaxUploadSpeed
+	}
+	if r.persist.StreamCacheSize == 0 {
```

Comment:
This prevents the user from setting the StreamCacheSize to 0. I realize that we currently don't allow users to set the stream cache size to zero, but if we enable that in the future users are going to run into an issue where they are unable to save a stream cache with size zero, it'll always load and reset to 2.

This is a similar issue for MaxDownloadSpeed and MaxUploadSpeed. It happens here that the defaults are also zero, and I don't see us ever changing those defaults (hard to imagine a sane default that would be some other value), but it's still clunky code that works due to unique circumstances rather than being inherently correct code.

The proper way to do this upgrade would be to increase the version of the renter persist. There's a metadata object that's currently got the version set to v0.4. What we would want to do is update that to v1.3.3. This would cause legacy renter persist to fail to load with a 'bad version' error, in which case you'd want to try again with the legacy version. If that succeeds, you then know to set these new values to their defaults. That means that instead of checking for the empty value, you know that you are only setting these fields for the user at startup if they are doing a genuine upgrade from an earlier version.

I think I'm in favor of switching to a version upgrade here. I know it's not strictly necessary, but taking a shortcut might lead to issues later. But this is also a really simple persistence upgrade to perform, and a good way to get your feet wet with our persistence system.